applications/dashboard/controllers/api/ThemesApiController.php

```diff
@@ -117,273 +60,203 @@ private function contentTypeByAsset(string $assetKey): string {
-     * Get theme asset.
+     * Create new theme.
      *
-     * @param string $id The unique theme key or theme ID (ex: keystone).
-     * @param string $assetKey Unique asset key (ex: header, footer, fonts, styles)
-     *        Note: assetKey can be filename (ex: header.html, styles.css)
-     *              in that case file content returned instaed of json structure
-     * @link https://github.com/vanilla/roadmap/blob/master/theming/theming-data.md#api
-     *
-     * @return array|Data
+     * @param array $body Array of incoming params.
+     *        fields: name (required)
+     * @return array
      */
-    public function get_assets(string $id, string $assetKey) {
-        $this->permission();
+    public function post(array $body): array {
+        $this->permission("Garden.Settings.Manage");
 
-        $content = $this->getAssetData($id, $assetKey);
-        $contentType = $this->contentTypeByAsset($assetKey);
-        $result = new Data($content);
-        return $result->setHeader("Content-Type", $contentType);
-    }
+        $in = $this->themePostSchema('in')->setDescription('Create new custom theme.');
 
-    /**
-     * Result theme schema
-     *
-     * @param string $type
-     * @return Schema
-     */
-    private function themeResultSchema(string $type = 'out'): Schema {
-        $schema = $this->schema(
-            Schema::parse([
-                'type:s',
-                'themeID:s',
-                'assets?' => $this->assetsSchema()
-            ]),
-            $type
-        );
-        return $schema;
+        $out = $this->themeResultSchema('out');
+
+        $body = $in->validate($body);
+
+        $normalizedTheme = $this->themeModel->postTheme($body);
+
+        $theme = $out->validate($normalizedTheme);
+        return $theme;
     }
 
+
     /**
-     * Get 'assets' schema
+     * Update theme name by ID.
      *
-     * @return Schema
+     * @param int $themeID Theme ID
+     * @param array $body Array of incoming params.
+     *        fields: name (required)
+     * @return array
      */
-    private function assetsSchema(): Schema {
-        $schema = Schema::parse([
-            "header?" => new InstanceValidatorSchema(HtmlAsset::class),
-            "footer?" => new InstanceValidatorSchema(HtmlAsset::class),
-            "variables?" => new InstanceValidatorSchema(JsonAsset::class),
-            "fonts?" => new InstanceValidatorSchema(FontsAsset::class),
-            "scripts?" => new InstanceValidatorSchema(ScriptsAsset::class),
-            "styles:s?",
-            "javascript:s?",
-            "logo?" => new InstanceValidatorSchema(ImageAsset::class),
-            "mobileLogo?" => new InstanceValidatorSchema(ImageAsset::class),
-        ])->setID('themeAssetsSchema');
-        return $schema;
+    public function patch(int $themeID, array $body): array {
+        $this->permission("Garden.Settings.Manage");
+        $in = $this->themePostSchema('in')->setDescription('Update theme name.');
```

Comment:
I don't think these descriptions actually do anything anymore (Because it's all in openapi interfaces). We should be able to remove it.